nni/compression/experiment/config/vessel.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import base64
+import io
+from dataclasses import dataclass, asdict
+from pathlib import Path
+from typing import Any, Callable, Optional, Tuple, Union
+from typing_extensions import Literal
+
+import torch
+from torch import Tensor
+from torch.nn import Module
+from torch.optim import Optimizer
+
+from nni.algorithms.compression.v2.pytorch.utils.constructor_helper import OptimizerConstructHelper
+from nni.common import dump, load
+from nni.experiment.config.base import ConfigBase
+
+
+@dataclass(init=False)
+class CompressionVessel(ConfigBase):
+    model: Union[str, bytes]
+    finetuner: Union[str, bytes]
+    evaluator: Union[str, bytes]
+    dummy_input: Union[str, bytes]
+    trainer: Union[str, bytes, Literal['null'], None]
+    optimizer_helper: Union[str, bytes, Literal['null'], None]
+    criterion: Union[str, bytes, Literal['null'], None]
+    device: str
+
+    def __init__(self,
+                 model: Module,
+                 finetuner: Callable[[Module], None],
+                 evaluator: Callable[[Module], float],
+                 dummy_input: Tensor,
+                 trainer: Optional[Callable[[Module, Optimizer, Callable[[Any, Any], Any]], None]],
+                 optimizer_helper: Union[Optimizer, OptimizerConstructHelper, None],
+                 criterion: Optional[Callable[[Any, Any], Any]],
+                 device: str):
+        # should be wrote as override __init__
+        self.model = dump(model) if not isinstance(model, str) else model
```

Comment:
Dump now guarantees str.

Reply:
str means it's a dumped model, so don't dump again.

Reply:
What I'm trying to say here is that the return value of `nni.dump` must be a str, rather than `str | bytes`. The change is recently introduced.